experiments/overparameterized_broadness.py

```diff
@@ -0,0 +1,70 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.optim as optim
+
+torch.manual_seed(0)
+
+class Net_54(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.fc1 = nn.Linear(2, 5)
+        self.fc2 = nn.Linear(5, 4)
+        self.fc3 = nn.Linear(4, 1)
+
+    def forward(self, x):
+        # F.relu doesn't work here (best loss is 0.17).
```

Comment:
Thanks. That... kinda makes sense, but then I'm surprised [silu](https://pytorch.org/docs/stable/generated/torch.nn.SiLU.html) works better (or at least faster) than [sigmoid](https://pytorch.org/docs/stable/generated/torch.nn.Sigmoid.html). Silu also maps small numbers to 0 and big numbers to themselves, I would guess the main advantage it has over relu here is that it's at least capable of generating negative numbers?

I played around with https://dcato98.github.io/playground/ a bit to see if I could get a feel for what's going on, but it always uses tanh for the final output on classification problems and doesn't have anything like the xor dataset for regression. Plus the input is in [-1, 1] instead of {0, 1} which seems like it could make a big difference here. And it doesn't have silu.

Reply:
You are probably correct here. It all depends on the landscape of the loss functions image space. Different non- linearitys enable different solutions (like e.g. using negative activation). Looking at the latest research output for superpositions, this might also has a big impact on your result, since your models might inhibit different amounts of superpositions (like in a model of the same size might me able to encode more features depending on the activation function). Might be something we could look into after you are done with your current setup. For now I would suggest sticking with one activation function throughout your experiment.